policy/test_overlayfs.te

```diff
@@ -0,0 +1,127 @@
+#################################
+#
+# Policy for testing overlayfs file system support
+#
+
+# Domain for mounter process.
+type test_overlay_mounter_t, testdomain;
+domain_type(test_overlay_mounter_t)
+
+# Domain for client process.
+type test_overlay_client_t, testdomain;
+domain_type(test_overlay_client_t)
+
+type test_overlay_files_ro_t;
+files_type(test_overlay_files_ro_t)
+
+type test_overlay_files_rwx_t;
+files_type(test_overlay_files_rwx_t)
+
+type test_overlay_files_noaccess_t;
+files_type(test_overlay_files_noaccess_t)
+
+#
+# Mounter policy
+#
+allow test_overlay_mounter_t self:dir list_dir_perms;
+allow test_overlay_mounter_t self:file read_file_perms;
+allow test_overlay_mounter_t self:capability { sys_admin dac_override dac_read_search };
+
+unconfined_runs_test(test_overlay_mounter_t)
+
+kernel_search_proc(test_overlay_mounter_t)
+kernel_read_proc_symlinks(test_overlay_mounter_t)
+
+fs_relabelfrom_xattr_fs(test_overlay_mounter_t)
+fs_mount_xattr_fs(test_overlay_mounter_t)
+
+corecmd_shell_entry_type(test_overlay_mounter_t)
+corecmd_exec_bin(test_overlay_mounter_t)
+
+userdom_read_inherited_user_tmp_files(test_overlay_mounter_t)
+userdom_search_admin_dir(test_overlay_mounter_t)
+
+mount_exec(test_overlay_mounter_t)
+mount_rw_pid_files(test_overlay_mounter_t)
+
+files_search_all(test_overlay_mounter_t)
+
+fs_getattr_xattr_fs(test_overlay_mounter_t)
+
+selinux_getattr_fs(test_overlay_mounter_t)
+
+#
+# Mounter needs to be able to mount on r/o and r/w/x directories and files
+# relabelfrom/relableto can probably be removed if fscreate fixes work.
+#
+allow test_overlay_mounter_t test_overlay_files_ro_t:file mounton;
+allow test_overlay_mounter_t test_overlay_files_rwx_t:{dir file} mounton;
+allow test_overlay_mounter_t test_overlay_files_rwx_t:filesystem { mount relabelfrom relabelto };
+allow test_overlay_mounter_t test_overlay_files_rwx_t:file { relabelfrom relabelto };
```

Comment:
I don't think the above rule on :file is needed even with the current patches, since they use __vfs_setxattr_noperm() for the copy-up and therefore do not check whether the mounter can relabel the files.  That was supposed to be implicitly granted based on the fact that we allowed it :filesystem relabelfrom for the context mount.